lib/cext/ruby.h

```diff
@@ -824,9 +824,46 @@ MUST_INLINE int rb_jt_scan_args_1_star(int argc, VALUE *argv, const char *format
   return argc - 1;
 }
 
-int rb_scan_args(int argc, VALUE *argv, const char *format, ...);
+MUST_INLINE int rb_jt_scan_args(int argc, VALUE *argv, const char *format, VALUE *v1, VALUE *v2, VALUE *v3, VALUE *v4, VALUE *v5, VALUE *v6, VALUE *v7, VALUE *v8, VALUE *v9, VALUE *v10) {
+  switch (format[0]) {
+    case '0':
+      switch (format[1]) {
+        case ':':
+          return rb_jt_scan_args_0_hash(argc, argv, format, v1);
+        case '2':
+          return rb_jt_scan_args_02(argc, argv, format, v1, v2);
+      }
+      break;
+    case '1':
+      switch (format[1]) {
+        case '1':
+          return rb_jt_scan_args_11(argc, argv, format, v1, v2);
+        case '2':
+          return rb_jt_scan_args_12(argc, argv, format, v1, v2, v3);
+        case '*':
+          return rb_jt_scan_args_1_star(argc, argv, format, v1, v2);
+      }
+      break;
```

Comment:
Just `-constprop` on its own isn't enough. I'll merge this and then investigate further.